src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+circular dependency issues, however, it is better to acquire the locator directly
+from the plugin contract as above, which makes dependencies explicit.
+
+You can use the `.get()` method on the locators service to acquire the locator
+from the central registry:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const dashboardLocator = plugins.share.url.locators.get('DASHBOARD_APP_LOCATOR');
+  }
+}
+```
+
+Once have have a reference to the desired locator object, you can use that locator
+to navigate to various pages in Kibana or use it to build a URL string.
+
+The best way to use the locator is to use its `.navigateSync()` method to navigate
+in SPA way (without page reload):
+
+```ts
+dashboardLocator.navigateSync({ dashboardId: '123' });
+```
+
+The above will immediately navigate to the Dashboard app, to the dashboard with
+ID `123`.
+
+Another useful feature of locators is their ability to format a URL string of the
+destination. The best way to do it is using the `.getRedirectUrl()` method
+of locators.
+
+```ts
+const url = dashboardLocator.getRedirectUrl({ dashboardId: '123' });
+```
+
+The above will generate a *redirect* URL, which also preserves the serializable
+*location state* object, which will be passed to the destination app.
+
+Alternatively, for compatibility with old URL generators, you can also use the
+`.getUrl()` method of locators to generate just the final app destination URL
+string without the location state object.
+
+```ts
+const url = await dashboardLocator.getUrl({ dashboardId: '123' });
+```
+
+Note, in this process you lose the location state object, which effectively
+makes the final destination incorrect, if the app makes use of the location
+state object.
+
+For more ways working with locators see the `LocatorPublic` interface.
+
+
+## Short URLs
+
+The `share` plugin has Short URL service, which allows to create Kibana short
+URLs. The service with the same interface is available, both, on the server
+and browser.
+
+When creating a short URL it is possible to use a custom short *slug*, used
+to dereference the short URL. Also, a human-readable short slug can be
+automatically generated.
+
+First, to be able to create a short URL, one needs to get hold of a locator
+reference, for which the short URL will be created. (See above section on
+*Locators* for more info.) Lets say we want to create a short URL for the
+Dashboard application, for that, we first acquire reference to the dashboard
+locator:
+
+```ts
+class MyPlugin {
+  setup(core, plugins) {
+    const dashboardLocator = plugins.dashboard.locator;
+  }
+}
+```
+
+Now you can create a short URL using your locator and the short URL service:
+
+```ts
+const shortUrls = plugins.share.url.shortUrls.get(null);
+const url = await shortUrls.create({
+  locator,
+  params: {
+    dashboardId: '123',
+  }
+});
+```
+
+You can make the short URL slug human-readable by specifying the
+`humanReadableSlug` flag:
```

Comment:
silly question: what happens when both `slug: "cool-slug"` and `humanReadableSlug: true` are provided?